app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/GeneralNavigationTest.kt

```diff
@@ -0,0 +1,135 @@
+    val dbManagement: DatabaseManagement = DummyDatabaseManagement(DummyDatabaseService())
+
+    @Before
+    fun setup() {
+        hiltRule.inject()
+        Intents.init()
+    }
+
+    @After
+    fun cleanUp() {
+        Intents.release()
+        activityScenarioRule.scenario.close()
+    }
+
+
+    @Test // this one used to cause trouble
+    fun navigateToDisplayDatasetAndThenNavigateUpGoesToDatasetsOverview() {
+        navigateToDisplayDataset()
+        assertCurrentFragmentIsCorrect(R.id.displayDatasetFragment)
+        onView(withContentDescription("Navigate up")).perform(click());
+        assertCurrentFragmentIsCorrect(R.id.datasetsOverviewFragment)
+
+    }
+
+    @Test
+    fun createDatasetAndNavigateUpGoesToDatasetsOverview() {
+        onView(withId(R.id.datasetsOverviewFragment)).perform(click())
+        onView(withId(R.id.createDatasetButton)).perform(click())
+        assertCurrentFragmentIsCorrect(R.id.categoriesEditingFragment)
+        onView(withContentDescription("Navigate up")).perform(click());
+        assertCurrentFragmentIsCorrect(R.id.datasetsOverviewFragment)
+    }
+
+    @Test
+    fun openingNavigationAndClickingOnLoginWorks() {
+        onView(withId(R.id.drawer_layout)).perform(DrawerActions.open());
+        onView(withId(R.id.drawer_layout)).check(matches(isOpen()));
+        onView(withId(R.id.googleSignInWidget)).perform(click())
+        assertCurrentFragmentIsCorrect(R.id.googleSignInWidget)
+    }
+
+    @Test
+    fun bottomNavigationCanNavigateToLearning() {
+        onView(withId(R.id.learningDatasetSelectionFragment)).perform(click())
+        assertCurrentFragmentIsCorrect(R.id.learningDatasetSelectionFragment)
+    }
+
+    @Test
+    fun bottomNavigationCanNavigateToDatasetEditing() {
+        onView(withId(R.id.learningDatasetSelectionFragment)).perform(click())
+        assertCurrentFragmentIsCorrect(R.id.learningDatasetSelectionFragment)
+    }
+
+    @Test
+    fun buttonOnHomeToLearningWorks() {
+        onView(withId(R.id.startLearningButton)).perform(click())
+        assertCurrentFragmentIsCorrect(R.id.learningDatasetSelectionFragment)
+    }
+
+    private fun navigateToDisplayDataset() {
+        onView(withId(R.id.datasetsOverviewFragment)).perform(click())
+        Espresso.onView(ViewMatchers.withId(R.id.DatasetList_list))
+            .perform(
+                RecyclerViewActions.actionOnItemAtPosition<DatasetListRecyclerViewAdapter.ListItemViewHolder>(
+                    0,
+                    ViewActions.click()
+                )
+            )
+    }
+
+    private fun assertCurrentFragmentIsCorrect(expected: Int) {
+        activityScenarioRule.scenario.onActivity {
+            var currentFragmentContainer =
+                it.supportFragmentManager.findFragmentById(R.id.nav_host_fragment_container)
+            val currentFragment = currentFragmentContainer?.findNavController()?.currentDestination
+            assert(currentFragment?.id == expected)
```

Comment:
A Hamcrest `assertThat` would be nice there.

Reply:
done thanks